### PR TITLE
Add ENABLE_TESTING to enable to build unit tests optionally

### PR DESCRIPTION
**Pull request by laura-ding**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -506,6 +511,13 @@ macro(nebula_link_libraries target)
     )
 endmacro(nebula_link_libraries)
 
+function(nebula_add_subdirectory dir_name)
+    if ((NOT ENABLE_TESTING) AND (${dir_name} STREQUAL test))
```
Comment (yixinglu) [inline CMakeLists.txt]: I think it's better not to ignore the entire `test` directory, but rather to provide a macro like `nebula_add_test` to ignore a single UT.

Comment (laura-ding) [reply CMakeLists.txt]: In addition to binary, there are also objects. We specify the `test` directory as the directory of tests

Comment (yixinglu) [reply CMakeLists.txt]: Do all tests have to be placed in the test directory? How to disable one or more of these tests ?

Comment (dutor) [reply CMakeLists.txt]: @yixinglu To assume that all unit tests are placed in _**test**_ directory is not a strong assumption.

For the time being, we have not demands to disable the building of individual tests.

Comment (yixinglu) [reply CMakeLists.txt]: Got it!

**Comment by dutor** [comment]
jenkins go

**Pull request merged**